src/hooks/useIsVisible.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useRef, useState } from 'react'
+import scrollTopStore from '../stores/scrollTop'
+
+/**
+ * Determines if an element is partially visible in the viewport.
+ *
+ * @param [initialValue=false] - The initial visibility state of the element.
+ * @returns An array containing the visibility state and the ref object of the element to check.
+ */
+const useIsVisible = <T extends HTMLElement>(initialValue = false): [boolean, React.RefObject<T>] => {
+  const scrollTop = scrollTopStore.useState()
```

Comment:
You can also use the reactMinistore's `useEffect` directly:

```js
scrollTopStore.useEffect(checkVisibility)
```

Reply:
Very cool! Retrieving the `scrollTop` value only to use it as a hook dependency felt awkward, but I didn't realize `scrollTopStore.useEffect` existed. That's much nicer. Refactored: 84ce5aa6f6d67215bf3b1875c73a60aa29ab6612

Reply:
Yeah, there are a ton of React state management libraries out there, but honestly I've gotten so much mileage out of rolling my own [ministore](https://github.com/cybersemics/em/blob/2793615f9c268725f3ae9688ea1fa5250381a86a/src/stores/ministore.ts) and [react-ministore](https://github.com/cybersemics/em/blob/2793615f9c268725f3ae9688ea1fa5250381a86a/src/stores/react-ministore.ts) that are a mere 150 LOC combined.

Ministores can even be [composed](https://github.com/cybersemics/em/blob/2793615f9c268725f3ae9688ea1fa5250381a86a/src/components/LayoutTree.tsx#L89-L93).